gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateWebLoginImageConfig.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 OpsMX, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config;
+
+import com.netflix.spinnaker.gate.services.LoginImageService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@ConditionalOnExpression("${security.login.image.enabled:false}")
+@Configuration
+@EnableConfigurationProperties(GateLoginImageConfigProperties.class)
+public class GateWebLoginImageConfig implements WebMvcConfigurer {
+
+  private static final Logger log = LoggerFactory.getLogger(GateWebLoginImageConfig.class);
+
+  @Value("${security.login.image.resourcepath:/opt/spinnaker/config/}")
+  private String imageResourcePath;
+
+  @Value("${security.login.image.url:image.png}")
```

Comment:
Avoid the use of `@Value` annotations. Put these in a `@ConfigurationProperties` class.

Reply:
Removed @value and moved to @@ConfigurationProperties class